### PR TITLE
Update/ux reloading web pages

### PR DESCRIPTION
**Pull request by osala-eng**
## Description

Please include a summary of the change, e.g. what the new feature # is and/or what bug # it fixes. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes/Implements #(issue/feature)
Fixes page goes to login page on page reload even if you have already logged in.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [X] Tests
- [ ] Other chores such as maintenance

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration by modifying the list below.

- [ ] Test A
- [ ] Test B
- [ ] Test C

**Test Configuration**:

Please describe the test setup. List them below as bullet points.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] The commit message follows the convention of this project

### CODE REVIEW

**Review by janaka — COMMENTED** [review]
As discussed, remove the auth autologin and auth session cache decorators and implement the needed logic inside the methods.

Handle cookie encrypt secret validation failed in setup, don't throw an error, render a friendly _Docq failed to start type_ message.

**Comment by osala-eng** [comment]
Added a friendly UI to show when `docq` fails to initialize.

image

**Comment by janaka** [comment]
> Added a friendly UI to show when `docq` fails to initialize.
> 
> image

Can you change the copy to: "Something went wrong starting Docq."

**Comment by osala-eng** [comment]
> Can you change the copy to: "Something went wrong starting Docq."

Okay

File: web/utils/layout.py [hunk 1]
```diff
@@ -266,6 +268,7 @@ def public_access() -> None:
 
 def auth_required(show_login_form: bool = True, requiring_admin: bool = False, show_logout_button: bool = True) -> bool:
     """Decide layout based on current user's access."""
+    handle_set_cached_session_configs()
     auth = get_auth_session()
```
Comment (janaka) [inline web/utils/layout.py:272]: Are all these public functions for caching needed? it's bit confusing follow the logic. Maybe you need an internal set and get but one handler should work? 

The typical pattern for caching is to handle it on the getter.

if obj not in cache: 
   get from source
   set cache
return value from cache

File: source/docq/support/auth_utils.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+        return tuple(data_[0]), data_[1]
+    except Exception as e:
+        log.error("Failed to decrypt auth data: %s", e)
+        return None
+
+
+def _update_auth_expiry(session_id: str) -> None:
+    """Update the auth expiry time."""
+    try:
+        cached_sessions[session_id] = cached_sessions[session_id]
```
Comment (janaka) [inline source/docq/support/auth_utils.py:153]: ?

File: source/docq/support/auth_utils.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+def generate_session_id(length: int = 32) -> str:
+    """Generate a secure and unique session_id."""
+    id_ = token_hex(length // 2)
+    hmac_ = _create_hmac(id_)
+    session_data[hmac_] = id_
+    return hmac_
+
+
+def _set_session_id(session_id: str) -> None:
```
Comment (janaka) [inline source/docq/support/auth_utils.py:103]: rename to `_set_cookie_session_id`. same for `_get*`

File: source/docq/support/auth_utils.py [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        system_settings = get_organisation_settings(org_id=org_id, key=SystemSettingsKey.ENABLED_FEATURES)
+        if system_settings: # Only enable feature when explicitly enabled (dafault to Disabled)
+            return FeatureType.AUTO_LOGIN.name in system_settings
+        return False
+    except Exception as e:
+        log.error("Failed to check if auto login is enabled: %s", e)
+        return False
+
+
+def cache_session_state_configs(*args: tuple, **kwargs: dict[str, Any]) -> None:
```
Comment (janaka) [inline source/docq/support/auth_utils.py:172]: I think rename this to `_set_cache_auth_session` and this is what gets called in the `get_auth_session()` if there's a cache miss.

**Comment by janaka** [comment]
I left a few comments but feel like the structure is still off. Some of it is just confusion caused by naming conventions. But also cache setting and getting seems a bit spaghetti which could accidently cause a critical security regression in the future. That's my concern. 

Maybe it's easier for me to look dive into the code a bit deeper first before you make any changes based on my comments.

So please hold off making any adjustments until I say so. 🙏🏾

**Review by janaka — APPROVED** [review]

**Pull request merged**